kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/wifi/WpaSupplicantManager.java

```diff
@@ -181,4 +198,60 @@ public static String getWpaSupplicantConfigFilename(String ifaceName) {
 		}
 		return sb.toString();
 	}
+	
+	private static String privGetWpaSupplicantConfigFilename(String ifaceName) {
+		StringBuilder sb = new StringBuilder();
+		if (OS_VERSION.equals(KuraConstants.Intel_Edison.getImageName() + "_" + KuraConstants.Intel_Edison.getImageVersion() + "_" + KuraConstants.Intel_Edison.getTargetName())) {
+			sb.append("/etc/wpa_supplicant/wpa_supplicant.conf");
+		} else {
+			sb.append("/tmp/wpa_supplicant-").append(ifaceName).append(".conf");
+		}
+		return sb.toString();
+	}
+	
+	private static File generateSupplicantConfigFile(String ifaceName, Password passkey, WifiSecurity wifiSecurity) throws KuraException {
+		File retConfigFile = new File(privGetWpaSupplicantConfigFilename(ifaceName));
+		File configFile = new File(getWpaSupplicantConfigFilename(ifaceName));
+		if(!configFile.exists()) {
+			throw KuraException.internalError("Config file does not exist: " + configFile.getAbsolutePath());
+		}
+		BufferedReader br = null;
+		PrintWriter pw = null;	
+		try {
+			br = new BufferedReader(new FileReader(configFile));
+			pw = new PrintWriter(new FileWriter(retConfigFile));
```

Comment:
@ibinshtok Same as before, I think that the FileReader and the FileWriter should be managed and closed.